<!-- Launch node for inverse kinematics using the Vive hand controllers -->
<launch>

    <!-- Arguments -->
    <arg name="arm_ratio" default="1.0"/>
    <arg name="velocity_linear_max" default="0.3"/>
    <arg name="velocity_angular_max" default="0.4"/>
    <arg name="speed_fraction" default="0.2"/>
    <arg name="calibration_time" default="1.0"/>
    <!-- best weight values are position: 100.0, orientation: 1.0 -->
    <arg name="position_weight" default="100.0"/>
    <arg name="orientation_weight" default="1.0"/>

    <!-- Start VR Controller Node -->
    <node name="vr_hand_controllers" pkg="pepper_vr_teleop" type="vr_hand_controllers.py" output="screen">
        <param name="arm_ratio" value="$(arg arm_ratio)"/>
        <param name="velocity_linear_max" value="$(arg velocity_linear_max)"/>
        <param name="velocity_angular_max" value="$(arg velocity_angular_max)"/>
        <param name="speed_fraction" value="$(arg speed_fraction)"/>
        <param name="calibration_time" value="$(arg calibration_time)"/>
        <param name="position_weight" value="$(arg position_weight)"/>
        <param name="orientation_weight" value="$(arg orientation_weight)"/>
        <!-- <param name="yaw_offset" value="0.0"/> -->
    </node>

</launch>
